### DISCUSSION TITLE
Welcome to Derw

### DISCUSSION THREAD

**Discussion by eeue56** [eeue56/derw#14] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

If you wanna reach out to someone directly, try [Twitter](https://twitter.com/derwlang)

**Comment by danneu** [comment]
This project looks good. After 5 years of Elm, I basically want an Elm + TEA abstraction but with the ability to call Javascript synchronously from my app code. 

Most of my runtime errors  in Elm come from moving Json.Encode.Value across ports, and sometimes I'm just doing that to round trip an Elm value into JS to use a JS library function on it. Yet presumably the async FFI is to reduce runtime exceptions, so it's been something I'm willing to give up, but it's also something that Derw solves because the Typescript I call from Derw can see my Derw types and vice versa which isn't something you get with Elm.

I used Derw last night to rewrite an app I had just started in Elm and I had a good experience. Unlike most one man langs, I like that you already have tooling set up like VS Code formatting and the lang server. The Typescript output is also so simple that I keep it open in another tab to keep an eye on what's going on.

Good work. I've read your posts on Substack and I recognize your username from Elm over the years. I'm excited to see this progress!

**Reply by eeue56** [reply]
Awesome, glad to hear it! Let me know if you run into any bugs or problems and I can help you out!
